paper.md

```diff
@@ -2,11 +2,77 @@
+were developed and in which they are being reproduced.
+
+For example, consider a researcher working in a Windows environment
+leaves behind a bunch of Windows PowerShell [@payette2006windows] scripts. Now, if some other researcher with access to a Linux machine only wants to run those scripts in correct order in their environment, it would become time taking and difficult. They will probably 
+end up spending a huge amount of time setting up VM's, finding the 
+appropriate OS and interpreting the correct order of execution of the steps, 
+hence making the process highly inefficent and time consuming, which 
+it does not need to be.
+
+![An end-to-end example of a workflow. On the left we have the 
+`.yml` file that defines the workflow. On the right, a pictorial 
+representation of it.](./figures/casestudy.png){#fig:casestudy}
+
+Through this paper, we aim to address the reproducibility problem [@goodman2016does] in research. We propose a methodology in which a complex experimental 
+workflow is decomposed into several discrete steps and each step executes in a
+separate container. Doing this way, makes the workflow platform independent. Although Software (Linux) containers [@javed2017linux] are a relatively old technology [@menage_adding_2007], it was not until recently, with the rise of Docker, that they entered mainstream territory [@bernstein_containers_2014]. Since then, this technology has transformed the way applications get deployed in shared infrastructures, with 25% of companies using this form of software deployment [@datadog_surprising_2018], and a market size projected to be close to 5B by 2023 [@marketsandmarkets_application_2018]. Docker has been the *de facto* container runtime, with other container runtimes such as Singularity [@kurtzer_singularity_2017], Charliecloud [@priedhorsky_charliecloud_2017] and Podman [@podmancommunity_containers_2019] having emerged. Since, these container runtimes are available for almost every well known operating systems and architectures, experiments can be reproduced easily using containerized workflows in almost any environment.
+
+At this point, one might think "*Why not use a single container for the entire workflow ?*". 
+There are scenarios where a single container image is not suitable for implementing workflows associated to complex application testing or validating scientific explorations 
+[@zheng_integrating_2015]. For example, a workflow might involve 
+executing two steps, both of them requiring conflicting versions of a 
+language runtime (e.g. Python 2.7 **and** Python 3.6). In this 
+scenario, users could resort to solving such conflicts with the use of 
+package managers [@package_managers_wiki], but this defeats the purpose of containers, which is to _not_ have to do this sort of thing inside a container. More 
```

Comment:
we usually don't reference wikipedia. Package managers are a well know piece of technology, so we can skip this